### ISSUE TITLE
Add BBEdit to 'Reveal Line in…'

### ISSUE THREAD

**Issue by tflo** [ForkIssues/Tracker#2029]
BBEdit comes with a command line tool, which should make this easy. The syntax is:

```shell
bbedit +<line number> <file path>
```

Thanks for considering,

– Tom

**Comment by tflo** [comment]
Any progress – or thoughts – on this?

In the meantime, I could help myself a little bit by creating a "Go to Line in BBEdit…" custom command in the file contextual menu:

```shell
bbedit +$1{text} "${file}"
```

This is a bit clumsy, since I have to enter the line number manually in the prompt (`$1{text}`). 

**But it points to another possible solution:**

Give us a variable `${file:line}`. The content of the variable would be the line number of the first selected line, or, if nothing is selected, the number of the line where the cursor is in[^fn1].

This would have the advantage that you wouldn't have to hardcode in additional editors.

[^fn1]: There is no visible cursor in the GUI, but the cursor position must be known internally, since I can click at any location in the text and then make a selection via Shift-ArrowKey.

**Comment by DanPristupov** [comment]
Can you try this build? https://cdn.fork.dev/prerelease/Fork-2.43.10.dmg

<img width="574" alt="Screenshot 2024-06-26 at 15 46 17" src="https://github.com/fork-dev/Tracker/assets/618115/6b469e40-0973-4b06-bb0d-b29ba5cbdfe3">

**Comment by tflo** [comment]
The Reveal works great, thank you! 😍

<img width="317" alt="image" src="https://github.com/fork-dev/Tracker/assets/1410282/25745089-cf31-4863-aa81-8b7117c208b5">

‌
---

However, the File Tree view is completely empty with this version:

<img width="710" alt="image" src="https://github.com/fork-dev/Tracker/assets/1410282/b0357158-d0da-4744-b02b-4cc82ea5c891">

‌

(no File Tree issues with the regular Fork version, 2.43.1)

**Comment by DanPristupov** [comment]
Great!

Can you send me the log file to support@fork.dev please? The log file is located at `%localappdata%\fork\logs\fork.log`

**Comment by tflo** [comment]
Sent.

Thanks again for the BBEdit addition!

**Comment by DanPristupov** [comment]
Thank you for the log. Can you try this one, please? https://cdn.fork.dev/prerelease/Fork-2.43.11.dmg

**Comment by tflo** [comment]
2.43.11:

File Tree is back, but the Fork context menu with the "Reveal in…" is gone (just the standard text context menu instead).

<img width="240" alt="image" src="https://github.com/fork-dev/Tracker/assets/1410282/98e781a2-a5f6-41ce-b6b6-2113fa274187">


The other views (Changes, Commit) still have the good context menu. 

[Fork.log](https://github.com/user-attachments/files/15991568/Fork.log)

**Comment by DanPristupov** [comment]
I think it always worked this way. The 'Reveal' context menu was never in the FileTree tab. It's only in Changes.

**Comment by tflo** [comment]
Oh, you are right. Somehow I thought it was in all views, but that's probably because I didn't use it that often without BBEdit in there 😉.

But if it's an easy thing to implement, it would be nice to have it there too 🙄. Or is there a specific reason why it's not there (or shouldn't be there)?

**Comment by tflo** [comment]
I guess I mixed it up with my above mentioned "Go to Line in BBEdit..." custom command in the file context menu, which also works in the Tree view, and I used it regularly there.

**Comment by DanPristupov** [comment]
> But if it's an easy thing to implement, it would be nice to have it there too 🙄. Or is there a specific reason why it's not there (or shouldn't be there)?

Yes, there is a reason. The File Tree tab doesn't show diff and it uses a basic text control. It even has default text context menu.

The other places (Summary and Changes tabs, History, Working directory) use the *diff* control which has customized menu which includes 'Reveal in <editor>' option.

**Comment by DanPristupov** [comment]
We just released Fork 2.44 with the ability to reveal changed line in BBEdit

**Comment by tflo** [comment]
> The File Tree tab doesn't show diff and it uses a basic text control. It even has default text context menu. The other places […] use the diff control which has customized menu which includes 'Reveal in ' option.

Yes, that is what I was referring to. 

Is there a reason why you don't use the diff control in the Tree view? The only useless menu item for the Tree view would be "Copy as Patch" (and _I think_ that could be conditionally suppressed without writing an extra custom menu).

And even if you don't suppress it, the default text menu has more "dead" menus anyway (e.g. why would I want a "Font" submenu in a context where I can't change the font?)

But the "Reveal in" items are (almost) as useful in the Tree view as they are in the Diff view, and in any case more useful than the standard text menu. At least for me, since I often used my custom "Go to line in BBEdit" file command in the Tree view as well.

But of course this is just a small side thing that came to mind in the given context.

---

> We just released Fork 2.44 with the ability to reveal changed line in BBEdit

Just installed it! Many thanks again for adding BBEdit (and also Xcode!).
